apps/cowswap-frontend/src/legacy/state/price/updater.ts

```diff
@@ -106,63 +137,156 @@ function isRefetchQuoteRequired(
   //  - The quote will expire soon
   if (wasQuoteCheckedRecently(quoteInformation.lastCheck)) {
     // Don't Re-fetch if it was queried recently
-    return false
+    return DONT_FETCH_QUOTES
   } else if (isLoading) {
     // Don't Re-fetch if there's another quote going on with the same params
     // It's better to wait for the timeout or resolution. Also prevents an issue of refreshing too fast with slow APIs
-    return false
+    return DONT_FETCH_QUOTES
   } else if (quoteInformation.fee) {
     // Re-fetch if the fee is expiring soon
-    return isExpiringSoon(quoteInformation.fee.expirationDate, RENEW_FEE_QUOTES_BEFORE_EXPIRATION_TIME)
+    const expiringSoon = isExpiringSoon(quoteInformation.fee.expirationDate, RENEW_FEE_QUOTES_BEFORE_EXPIRATION_TIME)
+    return expiringSoon ? FETCH_BEST_QUOTE : DONT_FETCH_QUOTES
   }
 
-  return false
+  return DONT_FETCH_QUOTES
 }
 
 export default function FeesUpdater(): null {
   const { chainId, account } = useWalletInfo()
-  const verifiedQuotesEnabled = useVerifiedQuotesEnabled(chainId)
-
-  const { independentField, typedValue: rawTypedValue, recipient } = useSwapState()
+  const refetchQuote = useRef(useRefetchQuoteCallback())
+  const setQuoteError = useSetQuoteError()
+  const isLoading = useIsBestQuoteLoading()
+  const isOnline = useIsOnline()
+  const quotesMap = useAllQuotes({ chainId })
   const {
     currencies: { INPUT: sellCurrency, OUTPUT: buyCurrency },
     currenciesIds: { INPUT: sellCurrencyId, OUTPUT: buyCurrencyId },
     parsedAmount,
   } = useDerivedSwapInfo()
+  const quoteInfo = useMemo(() => {
+    return quotesMap && sellCurrencyId ? quotesMap[sellCurrencyId] : undefined
+  }, [quotesMap, sellCurrencyId])
 
-  const enoughBalance = useEnoughBalanceAndAllowance({ account, amount: parsedAmount })
+  const quoteParams = useQuoteParams({
+    chainId,
+    account,
+    sellCurrency,
+    buyCurrency,
+    sellCurrencyId,
+    buyCurrencyId,
+    parsedAmount,
+  })
```

Comment:
Created an auxiliar hook to simplify a bit the logic of the updater. The code is long enough.